include/nlohmann/json.hpp

```diff
@@ -3342,16 +3342,18 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     @since version 1.0.0
     */
     template < typename ValueType, typename std::enable_if <
-                   !std::is_pointer<ValueType>::value&&
-                   !std::is_same<ValueType, detail::json_ref<basic_json>>::value&&
-                   !std::is_same<ValueType, typename string_t::value_type>::value&&
-                   !detail::is_basic_json<ValueType>::value
-                   && !std::is_same<ValueType, std::initializer_list<typename string_t::value_type>>::value
+                    std::conjunction<
+                        std::negation<std::is_pointer<ValueType>>,
+                        std::negation<std::is_same<ValueType, detail::json_ref<basic_json>>>,
+                        std::negation<std::is_same<ValueType, typename string_t::value_type>>,
+                        std::negation<detail::is_basic_json<ValueType>>,
+                        std::negation<std::is_same<ValueType, std::initializer_list<typename string_t::value_type>>>,
+
 #if defined(JSON_HAS_CPP_17) && (defined(__GNUC__) || (defined(_MSC_VER) && _MSC_VER >= 1910 && _MSC_VER <= 1914))
-                   && !std::is_same<ValueType, typename std::string_view>::value
+                        std::negation<std::is_same<ValueType, std::string_view>>,
 #endif
-                   && detail::is_detected<detail::get_template_function, const basic_json_t&, ValueType>::value
-                   , int >::type = 0 >
+                        detail::is_detected_lazy<detail::get_template_function, const basic_json_t&, ValueType>
+                    >::value, int >::type = 0 >
```

Comment:
IMHO, there is no much sense to use templated/typed/parametrized logic operations for all operands. For the same reason as above. Of course, it can backfire in the future, but it wouldn't hurt too much since these operands can be replaced pretty quickly.

Reply:
You could technically do something like:

```c++
std::conjunction<
    std::bool_constant<
        !std::is_pointer<ValueType>::value &&
        !std::is_same<ValueType, detail::json_ref<basic_json>>::value &&
        !std::is_same<ValueType, typename string_t::value_type>::value &&
        !detail::is_basic_json<ValueType>::value &&
        !std::is_same<ValueType, std::initializer_list<typename string_t::value_type>>::value &&
        // ...
    >,
    detail::is_detected_lazy<detail::get_template_function, const basic_json_t&, ValueType>
>
```

But I don't think it provides a big benefit, and it's less consistent.

Reply:
Yeah, actually, it's
https://github.com/nlohmann/json/blob/ae9cfb17bd960ae014b01ac62ecf22d72cde4ee5/include/nlohmann/json.hpp#L3345-L3357

You are right about consistency for sure. Does it give a big benefit - I don't know, I'm just hoping that it helps. But looking more at this library, probably a few more instantiations wouldn't be noticeable.